h/services/delete_user.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from h.events import AnnotationEvent
+from h.models import Annotation, Group
+from h import storage
+
+
+class UserDeleteError(Exception):
+    pass
+
+
+class DeleteUserService(object):
+    def __init__(self, request):
```

Comment:
We generally try to avoid services depending on requests, but the deletion logic relies on having an actual request in several ways so I kept the dependency.

Reply:
Yup: in this case, that looks fair. It looks like the things we need from the request are:

* the database session (for querying things and then deleting things)
* the `notify_after_commit` method (for saying an annotation has been deleted)
* the request itself (for passing into an `AnnotationEvent` to say the annotation has been deleted)

While I’d like to get rid of the request dependency here, it looks like that would involve some fairly significant rework of how `AnnotationEvent`s work, so I can see now isn’t the right time.